source/extensions/filters/http/common/compressor/compressor.h

```diff
@@ -0,0 +1,172 @@
+#pragma once
+
+#include "envoy/compressor/compressor.h"
+#include "envoy/extensions/filters/http/compressor/v3/compressor.pb.h"
+#include "envoy/http/filter.h"
+#include "envoy/stats/scope.h"
+#include "envoy/stats/stats_macros.h"
+#include "envoy/stream_info/filter_state.h"
+
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace Compressors {
+
+/**
+ * All compressor filter stats. @see stats_macros.h
+ * "total_uncompressed_bytes" only includes bytes from requests that were marked for compression.
+ * If the request was not marked for compression, the filter increments "not_compressed", but does
+ * not add to "total_uncompressed_bytes". This way, the user can measure the memory performance of
+ * the compression.
+ * "header_gzip" is specific to the gzip filter and is deprecated since it duplicates
+ * "header_compressor_used".
+ */
+#define ALL_COMPRESSOR_STATS(COUNTER)                                                              \
+  COUNTER(compressed)                                                                              \
+  COUNTER(not_compressed)                                                                          \
+  COUNTER(no_accept_header)                                                                        \
+  COUNTER(header_identity)                                                                         \
+  COUNTER(header_gzip)                                                                             \
+  COUNTER(header_compressor_used)                                                                  \
+  COUNTER(header_compressor_overshadowed)                                                          \
+  COUNTER(header_wildcard)                                                                         \
+  COUNTER(header_not_valid)                                                                        \
+  COUNTER(total_uncompressed_bytes)                                                                \
+  COUNTER(total_compressed_bytes)                                                                  \
+  COUNTER(content_length_too_small)                                                                \
+  COUNTER(not_compressed_etag)
+
+/**
+ * Struct definition for compressor stats. @see stats_macros.h
+ */
+struct CompressorStats {
+  ALL_COMPRESSOR_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+class CompressorFilterConfig {
+public:
+  CompressorFilterConfig() = delete;
+  virtual ~CompressorFilterConfig() = default;
```

Comment:
nit: delete

Reply:
This is an abstract class. If I delete the destructor I get `non-virtual-dtor` compiler warning. And marking it as `delete` makes it impossible to store `CompressorFilterConfigSharedPtr` in the filter state.

Reply:
Sorry nm.